dpbench/benchmarks/l2_norm/l2_norm_sycl_native_ext/l2_norm_sycl/_l2_norm_kernel.hpp

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright 2022 Intel Corp.
+//
+// SPDX - License - Identifier : Apache 2.0
+///
+/// The files implements a SYCL-based Python native extension for the
+/// l2-norm benchmark.
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <stdlib.h>
+#include <type_traits>
+
+using namespace sycl;
+
+template <typename FpTy>
+void l2_norm_impl(queue Queue,
+                  size_t npoints,
+                  size_t dims,
+                  const FpTy *a,
+                  FpTy *d)
+{
+    // FpTy *sum = malloc_device<FpTy>(npoints, Queue);
+
+    Queue.fill<FpTy>(d, 0., npoints).wait();
+
+    Queue
+        .submit([&](handler &h) {
+            h.parallel_for<class theKernel>(range<1>{npoints}, [=](id<1> myID) {
+                size_t i = myID[0];
+                for (size_t k = 0; k < dims; k++) {
+                    d[i] += a[i * dims + k] * a[i * dims + k];
```

Comment:
Minor nit, but can we use 2D subscripts instead? Improves readability a bit.
```
d[i] += a[i][k]*a[i][k];
```

Reply:
1D array is used for now.